src/main/java/duke/Event.java

```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+/**
+ * Tasks that start at a specific time and ends at a specific time.
+ * 
+ * @author Tianqi-Zhu
+ */
+public class Event extends Task {
+    private boolean isTimeString;
+    private String timeString;
+    private LocalDate startDate; 
+    private LocalTime startTime;
+
+    public Event(String name, String time) {
+        super(name);
+        this.timeString = time;
+        this.isTimeString = true;
+    }
+
+    public Event(String name, LocalDate startDate) {
+        super(name);
+        this.startDate = startDate;
+        this.isTimeString = false; 
+    }
+
+    public Event(String name, LocalDate startDate, LocalTime startTime) {
+        super(name);
+        this.startDate = startDate;
+        this.startTime = startTime;
+        this.isTimeString = false; 
+    }
+
+    private String getTime() {
```

Comment:
Perhaps `getTime()` could be named as `getDateAndTime()` since the method will always return date, while time is only included if the user added more details to `Event`